packages/hagrid/hagrid/cli.py

```diff
@@ -2013,6 +2203,222 @@ def check_ip_for_ssh(
+    # TODO not sure why, but this call is waiting for prompt from user (have to enter q to proceed)
+    subprocess.check_call(  # nosec
+        cmd,
+        shell=True,
+    )
+
+
+# def extract_host_ips_aws(stdout: bytes) -> Optional[TypeList]:
+#     output = stdout.decode("utf-8")
+#     output_dict = json.loads(output)
+#     print(f'extract_host_ips_aws, output_dict: {output_dict}')
+#     host_ips: Optional[TypeList] = []
+#     if "Instances" in output_dict:
+#         for ec2_instance_metadata in output_dict["Instances"]:
+#             if "PublicIpAddress" in ec2_instance_metadata:
+#                 host_ips.append(ec2_instance_metadata["PublicIpAddress"])
+
+#     return host_ips
+
+
+def extract_instance_ids_aws(stdout: bytes) -> TypeList:
+    output = stdout.decode("utf-8")
+    output_dict = json.loads(output)
+    print(f"extract_instance_ids_aws, output_dict: {output_dict}")
+    instance_ids: TypeList = []
+    if "Instances" in output_dict:
+        for ec2_instance_metadata in output_dict["Instances"]:
+            if "InstanceId" in ec2_instance_metadata:
+                instance_ids.append(ec2_instance_metadata["InstanceId"])
+
+    return instance_ids
+
+
+def get_host_ips_given_instance_ids(
+    instance_ids: TypeList, timeout: int = 600, wait_time: int = 10
+) -> TypeList:
+    checks = int(timeout / wait_time)  # 10 minutes in 10 second chunks
+    instance_ids_str = " ".join(instance_ids)
+    cmd = f"aws ec2 describe-instances --instance-ids {instance_ids_str}"
+    cmd += " --query 'Reservations[*].Instances[*].{StateName:State.Name,PublicIpAddress:PublicIpAddress}'"
+    cmd += " --output json"
+    while checks > 0:
+        checks -= 1
+        print(
+            f"Inside get_host_ips_given_instance_ids, waiting for {wait_time} seconds"
+        )
+        time.sleep(wait_time)
+        desc_ec2_output = subprocess.check_output(cmd, shell=True)  # nosec
+        instances_output_json = json.loads(desc_ec2_output.decode("utf-8"))
+        print(f"instances_output_json: {instances_output_json}")  # TODO remove
+        host_ips: TypeList = []
+        all_instances_running = True
+        for reservation in instances_output_json:
+            for instance_metadata in reservation:
+                if instance_metadata["StateName"] != "running":
+                    all_instances_running = False
+                    break
+                else:
+                    host_ips.append(instance_metadata["PublicIpAddress"])
+        print(f"all_instances_running: {all_instances_running}")  # TODO remove
+        if all_instances_running:
+            print(f"Returning host_ips: {host_ips}")  # TODO remove
+            return host_ips
+        # else, wait another wait_time seconds and try again
+
+    return []
+
+
+def make_aws_ec2_instance(
+    ami_id: str, ec2_instance_type: str, key_name: str, security_group_name: str
+) -> TypeList:
+    # TODO incorporate "node_count" parameter to create multiple instances?
```

Comment:
node count is out of scope for now, we added it for azure because we needed it but lets leave it off for other targets for now.